dotnet/src/SemanticKernel.Abstractions/Memory/AIContextBehaviorsManager.cs

```diff
@@ -0,0 +1,144 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.AI;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.SemanticKernel;
+
+/// <summary>
+/// A container class for <see cref="AIContextBehavior"/> objects that manages their lifecycle and interactions.
+/// </summary>
+[Experimental("SKEXP0130")]
+public sealed class AIContextBehaviorsManager
+{
+    private readonly List<AIContextBehavior> _behaviors = new();
+
+    /// <summary>
+    /// Gets the list of registered <see cref="AIContextBehavior"/> objects.
+    /// </summary>
+    public IReadOnlyList<AIContextBehavior> Behaviors => this._behaviors;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="AIContextBehaviorsManager"/> class.
+    /// </summary>
+    public AIContextBehaviorsManager()
+    {
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="AIContextBehaviorsManager"/> class with the specified <see cref="AIContextBehavior"/> objects.
+    /// </summary>
+    /// <param name="aiContextBehaviors">The <see cref="AIContextBehavior"/> objects to add to the manager.</param>
+    public AIContextBehaviorsManager(IEnumerable<AIContextBehavior> aiContextBehaviors)
+    {
+        this._behaviors.AddRange(aiContextBehaviors);
+    }
+
+    /// <summary>
+    /// Adds a new <see cref="AIContextBehavior"/> objects.
+    /// </summary>
+    /// <param name="aiContextBehavior">The <see cref="AIContextBehavior"/> object to register.</param>
+    public void Add(AIContextBehavior aiContextBehavior)
+    {
+        this._behaviors.Add(aiContextBehavior);
+    }
+
+    /// <summary>
+    /// Adds all <see cref="AIContextBehavior"/> objects registered on the provided dependency injection service provider.
+    /// </summary>
+    /// <param name="serviceProvider">The dependency injection service provider to read <see cref="AIContextBehavior"/> objects from.</param>
+    public void AddFromServiceProvider(IServiceProvider serviceProvider)
+    {
+        foreach (var behavior in serviceProvider.GetServices<AIContextBehavior>())
+        {
+            this.Add(behavior);
+        }
+    }
```

Comment:
I would avoid adding this responsibility to the class and make this an extension method (only if extremely needed).

Personally I prefer this to be a constructor argument at the initialization, usage of serviceCollection normally happens in the initialization patterns.

```suggestion
    /// <summary>
    /// Adds all <see cref="AIContextBehavior"/> objects registered on the provided dependency injection service provider.
    /// </summary>
    /// <param name="serviceProvider">The dependency injection service provider to read <see cref="AIContextBehavior"/> objects from.</param>
    public void AddBehaviorsFromServiceProvider(IServiceProvider serviceProvider)
    {
        foreach (var behavior in serviceProvider.GetServices<AIContextBehavior>())
        {
            this.Add(behavior);
        }
    }
```

Reply:
I cannot make this a constructor param, since the one calling this won't be the one constructing the manager.  The manager will already exist on a thread and the user would just add behaviors.
The naming is to match the fact that we already have an Add method there and the class is for behaviors, so calling it AddBehaviors again doesn't make sense I think.
Why would we put this on an extension method when we own the class itself?